core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -102,11 +102,20 @@ class SparkHadoopUtil extends Logging {
         hadoopConf.set("fs.s3n.awsSecretAccessKey", accessKey)
         hadoopConf.set("fs.s3a.secret.key", accessKey)
       }
-      // Copy any "spark.hadoop.foo=bar" system properties into conf as "foo=bar"
       conf.getAll.foreach { case (key, value) =>
+        // Copy any "spark.hadoop.foo=bar" system properties into conf as "foo=bar"
         if (key.startsWith("spark.hadoop.")) {
           hadoopConf.set(key.substring("spark.hadoop.".length), value)
         }
+        // fix added for SPARK-13979
+	// Copy any "fs.swift2d.foo=bar" or "fs.swift.foo=bar" properties into conf 
+        else if (key.startsWith("fs.swift")){
+          hadoopConf.set(key, value)
+        }
+        // Copy any "fs.s3.foo=bar" or "fs.s3a.foo=bar" or "fs.s3n.foo=bar" properties into conf 
+        else if (key.startsWith("fs.s3")){
+          hadoopConf.set(key, value)
+        }
```

Comment:
@steveloughran  can u help me with some default setting for adl, wasb, oss, gs to test or just syntax for them so that i can decide on filter conditions